server/src/main/java/org/opensearch/index/ReplicationStats.java

```diff
@@ -29,17 +31,26 @@ public class ReplicationStats implements ToXContentFragment, Writeable {
     public long maxBytesBehind;
     public long maxReplicationLag;
     public long totalBytesBehind;
+    public long totalRejections;
+    public ShardId shardId;
 
-    public ReplicationStats(long maxBytesBehind, long totalBytesBehind, long maxReplicationLag) {
+    public ReplicationStats(ShardId shardId, long maxBytesBehind, long totalBytesBehind, long maxReplicationLag, long totalRejections) {
```

Comment:
The SR stats components were designed to support the cat segrep API and the stats tracker will fetch additional stats not required here, so there are some small changes required to do this:

To clarify what I'm suggesting is:

- Add ctor param for a new object SegmentReplicationPressureStats into NodeStats.java Where SegmentReplicationPressureStats is simply a map of of ShardId/rejection count and a summed total of rejections.
- Add a method to SegmentReplicationPressureService to fetch SegmentReplicationPressureStats that fetches what it needs from SegmentReplicationStatsTracker.  Likely just fetch its rejectionCount map and build the SegmentReplicationPressureStats object.  The existing `getStats` and `getStatsForShard` methods are fetching additional stats from the primary shards that we won't require here. 
- Inject segmentReplicationPressureService into NodeService to fetch the stats

Reply:
>a shard level breakdown from toXContent

We can't have shard level break down in `_node/stats` API . That can make the response humongous.  We will need to aggregate it over the node 

Initially we  also thought IndexingPressure and ShardIndexingPressure stats would be a good example for us to follow, but those are node level constructs which doesn't work on shard level data , unlike our use case. That is why we prefer injecting this in SegmentStats. This also gels well with existing `remote_store` and `segment_replication` blobs to add .

Reply:
Do we need to include the shard breakdowns at all if we have cat segrep & remote store APIs surfacing this data?  node stats can then show only the node aggregate?

If not - lets pls inject a function here instead of the pressure service itself.

Reply:
We don't need shard breakdowns . Can @Rishikesh1159 please take up the Segment Pressure service part , since he already has changes ready ?  Will limit the scope to remote store rejections here.